pyiron_base/project/generic.py

```diff
@@ -1185,27 +1185,27 @@ def remove_job(self, job_specifier, _unprotect=False):
         if isinstance(job_specifier, (list, np.ndarray)):
             for job_id in job_specifier:
                 self.remove_job(job_specifier=job_id, _unprotect=_unprotect)
-        else:
-            if not self.db.view_mode:
-                try:
-                    job = self.inspect(job_specifier=job_specifier)
-                    if job is None:
-                        state.logger.warning(
-                            "Job '%s' does not exist and could not be removed",
-                            str(job_specifier),
-                        )
-                    elif _unprotect:
-                        job.remove_child()
-                    else:
-                        job.remove()
-                except IOError as _:
-                    state.logger.debug(
-                        "hdf file does not exist. Removal from database will be attempted."
-                    )
-                    job_id = self.get_job_id(job_specifier)
-                    self.db.delete_item(job_id)
+            return
+        if self.db.view_mode:
+            raise EnvironmentError("copy_to: is not available in Viewermode !")
+        job = self.inspect(job_specifier=job_specifier)
+        if job is None:
+            state.logger.warning(
+                "Job '%s' does not exist and could not be removed",
+                str(job_specifier),
+            )
+            return
+        try:
+            if _unprotect:
+                job.remove_child()
             else:
-                raise EnvironmentError("copy_to: is not available in Viewermode !")
+                job.remove()
+        except IOError as _:
+            state.logger.debug(
+                "hdf file does not exist. Removal from database will be attempted."
+            )
+            self.db.delete_item(job.id)
```

Comment:
@jan-janssen You missed this line. It might not be called often but it is another redundant database query.

Reply:
Ok, I realised that the `inspect()` function also works when the file is corrupted https://github.com/pyiron/pyiron_base/pull/1327